debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/BinlogReader.java

```diff
@@ -378,18 +399,54 @@ protected void handleServerHeartbeat(Event event) {
     /**
      * Handle the supplied event that signals that an out of the ordinary event that occurred on the master. It notifies the slave
      * that something happened on the master that might cause data to be in an inconsistent state.
-     * 
+     *
      * @param event the server stopped event to be processed; may not be null
      */
     protected void handleServerIncident(Event event) {
-        logger.trace("Server incident: {}", event);
+        if (event.getData() instanceof EventDataDeserializationExceptionData) {
+            EventDataDeserializationExceptionData data = event.getData();
+
+            // TODO make it an option
+            EventDeserializationFailureHandlingMode deserializationFailureHandling = EventDeserializationFailureHandlingMode.FAIL;
+            EventHeaderV4 eventHeader = (EventHeaderV4) data.getCause().getEventHeader(); // safe cast, instantiated that ourselves
+
+            if(deserializationFailureHandling == EventDeserializationFailureHandlingMode.FAIL) {
+                // logging some additional context but not the exception itself, this will happen in handleEvent()
+
+                logger.error(
+                        "Error while deserializing binlog event at offset {}." + System.lineSeparator() +
+                        "Use the mysqlbinlog tool to view the problematic event: `mysqlbinlog --start-position={} --stop-position={} --verbose {}",
```

Comment:
Why not use a parameter for `System.lineSeparator()` so that we don't have to do string manipulation here and so that the message is clearer?

Reply:
Ok, good idea.

Reply:
Done now.